libs/application/templates/children-residence-change/src/fields/Overview/SignatureModal/ErrorMessage.tsx

```diff
@@ -1,21 +1,41 @@
-import React from 'react'
-import { useIntl } from 'react-intl'
+import React, { useState, useEffect } from 'react'
+import { MessageDescriptor, useIntl } from 'react-intl'
 import { Box, AlertMessage, Button } from '@island.is/island-ui/core'
 import { signatureModal } from '../../../lib/messages'
 
 interface ErrorMessageProps {
   onClose: () => void
+  errorCode?: number
 }
 
-// TODO: handle errors better when we have the support from the backend
-const ErrorMessage = ({ onClose }: ErrorMessageProps) => {
+const ErrorMessage = ({ onClose, errorCode }: ErrorMessageProps) => {
   const { formatMessage } = useIntl()
+  const [errorMessage, setErrorMessage] = useState<MessageDescriptor>(
+    signatureModal.defaultError.message,
+  )
+
+  useEffect(() => {
+    switch (errorCode) {
+      case 404:
+        setErrorMessage(signatureModal.error.noElectronicId)
+        break
+      case 400:
+        setErrorMessage(signatureModal.error.userCancelled)
+        break
+      case 409:
+        setErrorMessage(signatureModal.error.timeOut)
+        break
+      default:
+        setErrorMessage(signatureModal.defaultError.message)
+    }
+  }, [errorCode])
+
   return (
     <>
       <AlertMessage
         type="error"
```

Comment:
Wondering if we should be using a different type here if for example the user aborts. 
https://ui.devland.is/?path=/docs/alerts-alertmessage--info - we could use warning or info in that case